### ISSUE TITLE
Limiting results

### ISSUE THREAD

**Issue by mikt** [yajra/laravel-datatables#270]
Hi there,

I'm having troubles scaling my application.
I have an entity called tickets. In my application I need to show the open tickets, which (at the time of writing) is 35k rows.

When just selecting all those tickets, the application stops working (501 error).

My code is rather simple:
```
 $tickets = Ticket::whereStatus('open');
        if (isset($category) && $category != $allCategories) {
            $tickets = $tickets->whereCategory($category);
        }
        $tickets = $tickets->select('*')->limit('20')->get();
        return Datatables::of($tickets)
```
I set the limit just to check this was actually the error.

I would really love to use the search feature of data tables, so that you can search for all the rows in the database, is there any way around this?

So to sum up: The above code will only fetch 20 results, and the search part of datatables appears only to be working within those 20 results.

If I omit the limit I set, I reach the desired effect, however this prevents the application from running.

**Comment by yajra** [comment]
Use query builder instead of collection. You can do this by removing `get` on your query. No need to set the limit as this is automatically set by the package along with the pagination settings of datatables jquery plugin.

```php
        $tickets = Ticket::whereStatus('open');
        if (isset($category) && $category != $allCategories) {
            $tickets = $tickets->whereCategory($category);
        }
        return Datatables::of($tickets)
```

**Comment by yajra** [comment]
Using query builder and/or eloquent engine should limit the results. Thanks!
